pages/bookmarks.jsx

```diff
@@ -25,11 +32,44 @@ export default React.createClass({
   },
   componentDidMount() {
     // get IDs of user's bookmarked entries
-    this.setState({bookmarks: getBookmarks()});
+    this.setState({bookmarkedIds: getBookmarks()}, () => {
+      this.fetchData();
+    });
 
     // let's verify what the logged in status is
     user.verify(this.props.router.location, () => this.setState({ user }));
   },
+  fetchData() {
+    // We are about to make a new request, set loadingData to true.
+    this.setState({ loadingData: true });
+
+    // find out entry ids in the batch we are going to fetch
+    let ids = this.state.bookmarkedIds.slice(this.state.batchIndex*PROJECT_BATCH_SIZE, (this.state.batchIndex+1)*PROJECT_BATCH_SIZE);
+
+    let params = {
+      ids: ids.join(`,`),
+      page: 1
+    };
+
+    Service.entries
+      .get(params)
+      .then((data) => {
+        this.setState({
+          loadingData: false,
+          entries: this.state.entries.concat(this.sortEntriesByTimeBookmarked(data.results)),
+          batchIndex: this.state.batchIndex+1,
+          moreEntriesToFetch: (this.state.batchIndex+1)*PROJECT_BATCH_SIZE < this.state.bookmarkedIds.length
+        });
+      })
+      .catch((reason) => {
+        console.error(reason);
+      });
+  },
+  sortEntriesByTimeBookmarked(entries) {
+    return entries.sort((a,b) => {
+      return this.state.bookmarkedIds.indexOf(a.id) > this.state.bookmarkedIds.indexOf(b.id);
```

Comment:
probably worth lifting out:
```
let ids = this.state.bookmarkedIds;
let sorter = (a,b) => ids.indexOf(a.id) - ids.indexOf(b.id);
return entries.sort(sorter);
```
Also note that small rewrite: a sorting function returns three numerical values, not a boolean (nevative number for less than, 0 for equal, positive number for greater than)